internal/services/network/network_manager_ipam_pool_static_cidr_resource_test.go

```diff
@@ -0,0 +1,248 @@
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.ipAddressNumber(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r ManagerIpamPoolStaticCidrResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := staticcidrs.ParseStaticCidrID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Network.StaticCidrs.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r ManagerIpamPoolStaticCidrResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%[1]s
+
+resource "azurerm_network_manager_ipam_pool_static_cidr" "test" {
+  name             = "acctest-ipsc-%[2]d"
+  ipam_pool_id     = azurerm_network_manager_ipam_pool.test.id
+  address_prefixes = ["10.0.0.0/27"]
+  lifecycle {
+    ignore_changes = [number_of_ip_addresses_to_allocate]
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r ManagerIpamPoolStaticCidrResource) ipAddressNumber(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%[1]s
+
+resource "azurerm_network_manager_ipam_pool_static_cidr" "test" {
+  name                               = "acctest-ipsc-%[2]d"
+  ipam_pool_id                       = azurerm_network_manager_ipam_pool.test.id
+  number_of_ip_addresses_to_allocate = "16"
+  lifecycle {
+    ignore_changes = [address_prefixes]
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r ManagerIpamPoolStaticCidrResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_network_manager_ipam_pool_static_cidr" "import" {
+  name             = azurerm_network_manager_ipam_pool_static_cidr.test.name
+  ipam_pool_id     = azurerm_network_manager_ipam_pool_static_cidr.test.ipam_pool_id
+  address_prefixes = azurerm_network_manager_ipam_pool_static_cidr.test.address_prefixes
+  lifecycle {
+    ignore_changes = [number_of_ip_addresses_to_allocate]
+  }
+}
+`, r.basic(data))
+}
+
+func (r ManagerIpamPoolStaticCidrResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%[1]s
+
+resource "azurerm_network_manager_ipam_pool_static_cidr" "test" {
+  name             = "acctest-ipsc-%[2]d"
+  ipam_pool_id     = azurerm_network_manager_ipam_pool.test.id
+  address_prefixes = ["10.0.0.0/26", "10.0.0.128/27"]
+  lifecycle {
+    ignore_changes = [number_of_ip_addresses_to_allocate]
```

Comment:
Would an option to split this into two resources work better instead of using ignore_changes?

Reply:
Hi, I think it isn't necessary to split this into two separate resources, because it's the same resource type with the same function; only the IP address can be set from prefixes or a number.